doc/source/groupby.rst

```diff
@@ -946,11 +946,15 @@ that is itself a series, and possibly upcast the result to a DataFrame:
 
 .. warning::
 
-    In the current implementation apply calls func twice on the
+    * In the current implementation apply calls func twice on the
     first group to decide whether it can take a fast or slow code
     path. This can lead to unexpected behavior if func has
     side-effects, as they will take effect twice for the first
     group.
+    
+    * Apply should not apply functions to the group chunk. 
```

Comment:
this is not very clear here, its the *user* function that should not modify the group chunks.

Reply:
maybe instead
```
* for predictable results and performance, func should **not** mutate any of the group chunks
```

Reply:
This has been resolved.